public/app/features/query/components/QueryEditorRowTitle.tsx

```diff
@@ -2,37 +2,36 @@ import React, { useState } from 'react';
 import { css, cx } from '@emotion/css';
 import { DataQuery, DataSourceInstanceSettings, GrafanaTheme, TimeRange } from '@grafana/data';
 import { DataSourcePicker } from '@grafana/runtime';
-import { Icon, Input, stylesFactory, useTheme, FieldValidationMessage, TimeRangeInput } from '@grafana/ui';
+import { Icon, Input, FieldValidationMessage, TimeRangeInput, useStyles } from '@grafana/ui';
 import { selectors } from '@grafana/e2e-selectors';
-import { ExpressionDatasourceID } from '../../expressions/ExpressionDatasource';
+import { ExpressionDatasourceUID } from '../../expressions/ExpressionDatasource';
 
 export interface Props {
   query: DataQuery;
   queries: DataQuery[];
-  dataSourceName: string;
-  inMixedMode?: boolean;
   disabled?: boolean;
   timeRange?: TimeRange;
-  onTimeRangeChange?: (timeRange: TimeRange) => void;
+  dataSourceSettings: DataSourceInstanceSettings;
+  onChangeDataSourceSettings?: (instanceSettings: DataSourceInstanceSettings) => void;
+  onChangeTimeRange?: (timeRange: TimeRange) => void;
   onChange: (query: DataQuery) => void;
   onClick: (e: React.MouseEvent) => void;
   collapsedText: string | null;
 }
 
-export const QueryEditorRowTitle: React.FC<Props> = ({
-  dataSourceName,
-  inMixedMode,
-  disabled,
-  query,
-  queries,
-  onClick,
-  onChange,
-  onTimeRangeChange,
-  timeRange,
-  collapsedText,
-}) => {
-  const theme = useTheme();
-  const styles = getQueryEditorRowTitleStyles(theme);
+export const QueryEditorRowTitle: React.FC<Props> = (props) => {
```

Comment:
Maybe rename this to QueryEditorRowHeader? feels like it does more than the title :)